object_goslice_test.go

```diff
@@ -271,3 +271,138 @@ func TestGoSliceToString(t *testing.T) {
+		threshold      int
+		expectedMem    uint64
+		expectedNewMem uint64
+		errExpected    error
+	}{
+		{
+			name:      "should account for each value given a non-empty slice",
+			threshold: 100,
+			val: &objectGoSlice{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: &[]interface{}{
+					valueInt(99),
+					valueInt(99),
+					valueInt(99),
+					valueInt(99),
+				},
+			},
+			// overhead + values
+			expectedMem: SizeEmptyStruct + SizeInt*4,
+			// overhead + values
+			expectedNewMem: SizeEmptyStruct + SizeInt*4,
+			errExpected:    nil,
+		},
+		{
+			name:      "should account for each value given slice with a nil value",
+			threshold: 100,
+			val: &objectGoSlice{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: &[]interface{}{nil},
+			},
+			// overhead + null
+			expectedMem: SizeEmptyStruct + SizeEmptyStruct,
+			// overhead + null
+			expectedNewMem: SizeEmptyStruct + SizeEmptyStruct,
+			errExpected:    nil,
+		},
+		{
+			name:      "should account slice length over threshold",
+			threshold: 20,
+			val: &objectGoSlice{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: &[]interface{}{
+					valueInt(99),
+					valueInt(99),
+					valueInt(99),
+					valueInt(99),
+				},
+			},
+			// overhead + values
+			expectedMem: SizeEmptyStruct + SizeInt*4,
+			// overhead + values
+			expectedNewMem: SizeEmptyStruct + SizeInt*4,
+			errExpected:    nil,
+		},
+		{
+			name:      "should account for nested slices",
+			threshold: 100,
+			val: &objectGoSlice{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: &[]interface{}{
+					[]interface{}{
+						valueInt(99),
+						valueInt(99),
+						valueInt(99),
+						valueInt(99),
+					},
+				},
+			},
+			// overhead + (value + len("length") + "length".value + prototype + ints)
+			expectedMem: SizeEmptyStruct + (SizeEmptyStruct + 6 + SizeEmptyStruct + (SizeEmptyStruct + SizeEmptyStruct) + SizeNumber*4),
+			// overhead + (value + len("length") with string overhead + "length".value + prototype + ints)
+			expectedNewMem: SizeEmptyStruct + (SizeEmptyStruct + (6 + SizeString) + SizeEmptyStruct + (SizeEmptyStruct + SizeEmptyStruct) + SizeNumber*4),
+			errExpected:    nil,
+		},
+		{
+			name:      "should account for objectGoSlice", // treated as a objectGoSliceReflect
+			threshold: 100,
+			val: &objectGoSlice{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: &[]interface{}{
+					&objectGoSlice{
+						baseObject: baseObject{
+							val: &Object{runtime: vm},
+						},
+						data: &[]interface{}{
+							valueInt(99),
+							valueInt(99),
+							valueInt(99),
+							valueInt(99),
+						},
+					},
+				},
+			},
+			// overhead + nested overhead
+			expectedMem: SizeEmptyStruct + SizeEmptyStruct,
+			// overhead + nested overhead
+			expectedNewMem: SizeEmptyStruct + SizeEmptyStruct,
+			errExpected:    nil,
+		},
```

Comment:
[Q] Similar to the map Q of nested pre-parsed objects.

Reply:
This is also similar and I would use a pointer to a slice of interfaces rather than the internal objectGoSlice type. Again this test is correct but if we use a pointer it better represent what a user could do (where the user is us in baas eheh)